src/_avif.c

```diff
@@ -0,0 +1,932 @@
+
+    avifResult result;
+
+    if (!PyArg_ParseTuple(args, "y*si", &buffer, &codec_str, &max_threads)) {
+        return NULL;
+    }
+
+    if (strcmp(codec_str, "auto") == 0) {
+        codec = AVIF_CODEC_CHOICE_AUTO;
+    } else {
+        codec = avifCodecChoiceFromName(codec_str);
+    }
+
+    self = PyObject_New(AvifDecoderObject, &AvifDecoder_Type);
+    if (!self) {
+        PyErr_SetString(PyExc_RuntimeError, "could not create decoder object");
+        PyBuffer_Release(&buffer);
+        return NULL;
+    }
+
+    decoder = avifDecoderCreate();
+#if AVIF_VERSION >= 80400
+    decoder->maxThreads = max_threads;
+#endif
+#if AVIF_VERSION >= 90200
+    // Turn off libavif's 'clap' (clean aperture) property validation.
+    decoder->strictFlags &= ~AVIF_STRICT_CLAP_VALID;
+    // Allow the PixelInformationProperty ('pixi') to be missing in AV1 image
+    // items. libheif v1.11.0 and older does not add the 'pixi' item property to
+    // AV1 image items.
+    decoder->strictFlags &= ~AVIF_STRICT_PIXI_REQUIRED;
+#endif
+    decoder->codecChoice = codec;
+
+    result = avifDecoderSetIOMemory(decoder, buffer.buf, buffer.len);
+    if (result != AVIF_RESULT_OK) {
+        PyErr_Format(
+            exc_type_for_avif_result(result),
+            "Setting IO memory failed: %s",
+            avifResultToString(result)
+        );
+        avifDecoderDestroy(decoder);
+        PyBuffer_Release(&buffer);
+        PyObject_Del(self);
+        return NULL;
+    }
+
+    result = avifDecoderParse(decoder);
+    if (result != AVIF_RESULT_OK) {
+        PyErr_Format(
+            exc_type_for_avif_result(result),
+            "Failed to decode image: %s",
+            avifResultToString(result)
+        );
+        avifDecoderDestroy(decoder);
+        PyBuffer_Release(&buffer);
+        PyObject_Del(self);
+        return NULL;
+    }
+
+    self->decoder = decoder;
+    self->buffer = buffer;
+
+    return (PyObject *)self;
+}
+
+PyObject *
+_decoder_dealloc(AvifDecoderObject *self) {
+    if (self->decoder) {
+        avifDecoderDestroy(self->decoder);
+    }
+    PyBuffer_Release(&self->buffer);
+    Py_RETURN_NONE;
+}
+
+PyObject *
+_decoder_get_info(AvifDecoderObject *self) {
+    avifDecoder *decoder = self->decoder;
+    avifImage *image = decoder->image;
+
+    PyObject *icc = NULL;
+    PyObject *exif = NULL;
+    PyObject *xmp = NULL;
+    PyObject *ret = NULL;
+
+    if (image->xmp.size) {
+        xmp = PyBytes_FromStringAndSize((const char *)image->xmp.data, image->xmp.size);
+    }
+
+    if (image->exif.size) {
+        exif =
+            PyBytes_FromStringAndSize((const char *)image->exif.data, image->exif.size);
+    }
+
+    if (image->icc.size) {
+        icc = PyBytes_FromStringAndSize((const char *)image->icc.data, image->icc.size);
+    }
+
+    ret = Py_BuildValue(
+        "IIIsSSIS",
+        image->width,
+        image->height,
+        decoder->imageCount,
+        decoder->alphaPresent == AVIF_TRUE ? "RGBA" : "RGB",
```

Comment:
Nit: Omit `== AVIF_TRUE`

Reply:
I don't mind removing it, but let me ask - if this isn't the appropriate use case, then why does `AVIF_TRUE` exist at all?

Reply:
Hi Andrew,

I am afraid that there is a misunderstanding. `decoder->alphaPresent` is an expression with a "boolean" type (emulated with the `int` type in C). When testing a boolean expression, it is not necessary to compare it with the constant "true".

We can illustrate this point with an exaggerated example in C++. It is not necessary to write `if ((a > b) == true) {...}`. We can simply write `if (a > b) {...}`.

Reply:
It's been removed.